src/Features/Core/Portable/Completion/CompletionHelper.cs

```diff
@@ -289,23 +291,31 @@ private static int CompareExpandedItem(CompletionItem item1, PatternMatch match1
             }
```

Comment:
i hate the condition above this.  `if (a && b || c)` is so tough to read and reason about.   can we jsut break this out into different checks?